src/stitching/delegateToSchema.ts

```diff
@@ -39,547 +46,148 @@ export default async function delegateToSchema(
-      ...(variableDefinitions || []).filter(
-        variableDefinition =>
-          usedVariables.indexOf(variableDefinition.variable.name.value) !== -1,
-      ),
-      ...newVariableDefinitions
-    ],
-    selectionSet,
+  const rawRequest: Request = {
+    document: rawDocument,
+    variables: info.variableValues as Record<string, any>,
   };
 
-  const newDoc: DocumentNode = {
-    kind: Kind.DOCUMENT,
-    definitions: [operationDefinition, ...processedFragments],
-  };
+  const transforms = [
+    ...(options.transforms || []),
+    new AddArgumentsAsVariables(options.schema, args),
+    new FilterToSchema(options.schema),
+    new AddTypenameToAbstract(options.schema),
+    new CheckResultAndHandleErrors(info, options.fieldName),
+  ];
 
-  return newDoc;
-}
+  const processedRequest = applyRequestTransforms(rawRequest, transforms);
 
-function processRootField(
-  selection: FieldNode,
-  rootFieldName: string,
-  rootField: GraphQLField<any, any>,
-): {
-  selection: FieldNode;
-  variables: Array<{ arg: string; variable: string }>;
-} {
-  const existingArguments = selection.arguments || [];
-  const existingArgumentNames = existingArguments.map(arg => arg.name.value);
-  const allowedArguments = rootField.args.map(arg => arg.name);
-  const missingArgumentNames = difference(
-    allowedArguments,
-    existingArgumentNames,
-  );
-  const extraArguments = difference(existingArgumentNames, allowedArguments);
-  const filteredExistingArguments = existingArguments.filter(
-    arg => extraArguments.indexOf(arg.name.value) === -1,
-  );
-  const variables: Array<{ arg: string; variable: string }> = [];
-  const missingArguments = missingArgumentNames.map(name => {
-    // (XXX): really needs better var generation
-    const variableName = `_${name}`;
-    variables.push({
-      arg: name,
-      variable: variableName,
-    });
-    return {
-      kind: Kind.ARGUMENT,
-      name: {
-        kind: Kind.NAME,
-        value: name,
-      },
-      value: {
-        kind: Kind.VARIABLE,
-        name: {
-          kind: Kind.NAME,
-          value: variableName,
-        },
-      },
-    };
-  });
+  const errors = validate(options.schema, processedRequest.document);
+  if (errors.length > 0) {
+    throw errors;
+  }
 
-  return {
-    selection: {
-      kind: Kind.FIELD,
-      alias: null,
-      arguments: [...filteredExistingArguments, ...missingArguments],
-      selectionSet: selection.selectionSet,
-      name: {
-        kind: Kind.NAME,
-        value: rootFieldName,
-      },
-    },
-    variables,
-  };
+  if (options.operation === 'query' || options.operation === 'mutation') {
+    return applyResultTransforms(
+      await execute(
+        options.schema,
+        processedRequest.document,
+        info.rootValue,
+        options.context,
+        processedRequest.variables,
+      ),
+      transforms,
+    );
+  }
+
+  if (options.operation === 'subscription') {
+    // apply result processing ???
```

Comment:
I forgot to submit a comment from a couple weeks ago (sorry), and this code has changed a bit since then, but I think the question is still relevant: Is `// apply result processing ???` an important TODO?